[Term Entry] numpy built-in-functions: .mean() (#4483)
Codecademy/docs: content/numpy/concepts/built-in-functions/terms/mean/mean.md

---
Title: '.mean()'
Description: 'Computes the arithmetic mean along the specified axis.'
Subjects:
  - 'Computer Science'
  - 'Data Science'
Tags:
  - 'Data Structures'
  - 'Arrays'
  - 'Functions'
  - 'NumPy'
CatalogContent:
  - 'learn-python-3'
  - 'paths/data-science'
---

## Definition

The **`.mean()`** function calculates and returns the arithmetic mean, i.e. average, of an array of numbers. If the axis are specified, the average is taken over ha axis, and if not the mean is calculated across the flattened array.

## Syntax

```pseudo
numpy.mean(a, axis=None, dtype=None, out=None, keepdims=<no value>, *, where=<no value>)
```

`.mean()` provides the following arguments:

- `a`: It is the array of numbers for which the mean is wanted. This argument is mandatory, but all others are optional. If `a` is not a list, an error will be raised.
- `axis`: If the axis is given, the mean is computed over the given axis or axes. Otherwise, it is calculated over the flattened array.

> - `dtype`: It is the type which used when the mean is calculated. By default, [float64](https://numpy.org/doc/stable/reference/arrays.scalars.html#numpy.float64) is used for integers, and for floating point inputs, it is the same as the input `dtype`.

Reviewer: ```suggestion
- `dtype`: It is the data type for which the mean is calculated. By default, `float64` is used for integers, and for floating point inputs, it is the same as the input `dtype`.
```
I didn't understand what you mean by the second sentence here. You need to edit it.

> - `out`: This option allows to store the calculated mean directly in an already existing array, instead of creating a new array to hold the result. Also, this optional argument must have the same shape that is expected of the resulted mean.

Reviewer: ```suggestion
- `out`: It is an optional argument that allows storing the calculated mean in an existing array instead of creating a new array to hold the result. It must have the same shape that is expected of the resulting mean. The default value for `out` is `None`.
```

> - `keepdims`: This argument is **False** by default. Setting it to **True** specifies whether the dimensions that are reduced should be kept in the result.

Reviewer: ```suggestion
- `keepdims`: This argument is `False` by default. Setting it to `True` specifies whether the reduced dimensions should be kept in the result.
```

- `where`: It clarifies which elements should be included in the mean calcultion.

## Example

> The following example creates an array then uses a few `.mean()` operations to compute the mean of the elements.

Reviewer: ```suggestion
The following example creates an array and then uses the `.mean()` operation with different attributes to compute the mean of the elements.
```

## Syntax

```pseudo
numpy.mean(a, axis=None, dtype=None, out=None, keepdims=<no value>, *, where=<no value>)
```

`.mean()` provides the following arguments:

- `a`: Array containing numbers whose mean is desired. If a is not an array, a conversion is attempted. The `a` argument is required. All other arguments are optional.
- `axis`: Axis or axes along which the means are computed. The default is to compute the mean of the flattened array. If this is a tuple of ints, a mean is performed over multiple axes, instead of a single axis or all the axes as before.
- `dtype`: Type to use in computing the mean. For integer inputs, the default is [float64](https://numpy.org/doc/stable/reference/arrays.scalars.html#numpy.float64); for floating point inputs, it is the same as the input `dtype`. 
- `out`: Alternate output array in which to place the result. The default is **None**; if provided, it must have the same shape as the expected output, but the type will be cast if necessary.
- `keepdims`: If this is set to **True**, the axes which are reduced are left in the result as dimensions with size one. With this option, the result will broadcast correctly against the input array.

> - `where`: Elements to include in the mean.

Reviewer: Plagiarism: https://numpy.org/doc/stable/reference/generated/numpy.mean.html
